src/insights_client/__init__.py

```diff
@@ -95,6 +95,34 @@ def sorted_eggs(eggs):
         return [eggs[1], eggs[0]]
 
 
+def _remove_gpg_home(home):
+    """Clean GPG's temporary home directory at path 'home'.
+
+    :param home: Path to the GPG's temporary home.
+    :type home: str
+    :rtype: None
+    """
+    # Shut down GPG's home agent
+    shutdown_process = subprocess.Popen(
+        ["/usr/bin/gpgconf", "--homedir", home, "--kill", "all"],
+        stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+        universal_newlines=True,
+    )
+    stdout, stderr = shutdown_process.communicate()
+    if shutdown_process.returncode != 0:
+        log("Could not clean GPG home, got return code {rc}".format(rc=shutdown_process.returncode))
+        if stdout:
+            log(stdout)
+        if stderr:
+            log(stderr)
+
+    # Delete the temporary directory
+    try:
+        shutil.rmtree(home)
+    except Exception as exc:
+        log("Could not remove GPG home, got {exc}".format(exc=exc))
```

Comment:
should we still do not ignore the failures of `shutil.rmtree()` for now, to check whether anything wrong still happens on cleanup?

Reply:
I am a bit scared to do that, but it might be a good idea.